### ISSUE TITLE
KeyError when trying to lookup up timezone with reverse_timezone

### ISSUE THREAD

**Issue by charliesneath** [geopy/geopy#338]
I'm getting the following error when running the following code (with the correct usernames):

```
import json
from geopy.geocoders import Nominatim, GeoNames

lat = 40.6997716
lon = -73.9753359

geolocator = Nominatim(user_agent="test")
tz_lookup = GeoNames(user_agent="test", username="test")
tz = tz_lookup.reverse_timezone((lat, lon))
print tz
```

```
Traceback (most recent call last):
  File "set_timezone.py", line 9, in <module>
    tz = tz_lookup.reverse_timezone((lat, lon))
  File "/usr/local/lib/python2.7/dist-packages/geopy/geocoders/geonames.py", line 301, in reverse_timezone
    return self._parse_json_timezone(response)
  File "/usr/local/lib/python2.7/dist-packages/geopy/geocoders/geonames.py", line 304, in _parse_json_timezone
    return from_timezone_name(response["timezoneId"], raw=response)
KeyError: 'timezoneId'
```

**Comment by KostyaEsmukov** [comment]
Apparently the issue you're experiencing is due to the `username` not being accepted by GeoNames:

```
$ http 'http://api.geonames.org/timezoneJSON?lat=40.75376406311989&lng=-73.98489005863667&username=test'
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Cache-Control: no-cache
Connection: Keep-Alive
Content-Length: 117
Content-Type: application/json;charset=UTF-8
Date: Wed, 12 Dec 2018 20:17:05 GMT
Keep-Alive: timeout=5, max=100
Server: Apache/2.4.6 (CentOS) mod_jk/1.2.41 OpenSSL/1.0.1e-fips

{
    "status": {
        "message": "user account has not been confirmed. Check your email for the confirmation email.",
        "value": 10
    }
}
```

The error should be gone if you'd use a username which GeoNames considers valid. The same request with a correct username seems to work for me.

Although I'd still call it a bug in geopy, because it should've been a clearer `GeocoderAuthenticationFailure` exception instead of an obscure `KeyError`, so thanks for reporting!

**Comment by KostyaEsmukov** [comment]
Another issue I discovered during investigation is that the case of non-existing timezones is not handled properly by geopy. For example, at `lat=79&lng=80` GeoNames does return GMT/DST offsets, but doesn't return a timezone name (in the Olson DB format). A similar problem applies to GoogleV3 as well.

I started working on the fix, hopefully it'll be done by this weekend.

**Comment by charliesneath** [comment]
@KostyaEsmukov Interesting! I have already clicked through the email and confirmed my account name. Is there another account I have to make in addition to the one with GeoNames? I've double-checked and I'm using my correct username that has an existing GeoNames account.

Is it necessary to include the `user_agent`?

Thanks for looking into this.

**Comment by KostyaEsmukov** [comment]
> Is there another account I have to make in addition to the one with GeoNames?

I don't remember the exact steps. 

You may try to perform an http query manually and see what would GeoNames tell you there. The query I posted before might be used for that (just put your `username` value in place of `test`. The `http` command is https://httpie.org/, but if you don't have or want it, `curl` might be used instead. Something like this:

```
$ curl 'http://api.geonames.org/timezoneJSON?lat=40.75376406311989&lng=-73.98489005863667&username=YOUR_ACTUAL_USERNAME'
```

> Is it necessary to include the user_agent?

No, AFAICT GeoNames doesn't require custom user agent value (like Nominatim does). A `username` should be enough.

**Comment by KostyaEsmukov** [comment]
Fixed via #339.

Now the same code snippet would raise the following instead of a KeyError:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kostya/Code/_contrib/geopy/geopy/geocoders/geonames.py", line 311, in reverse_timezone
    self._call_geocoder(url, timeout=timeout)
  File "/Users/kostya/Code/_contrib/geopy/geopy/geocoders/geonames.py", line 329, in _parse_json_timezone
    self._raise_for_error(response)
  File "/Users/kostya/Code/_contrib/geopy/geopy/geocoders/geonames.py", line 323, in _raise_for_error
    raise GeocoderAuthenticationFailure(message)
geopy.exc.GeocoderAuthenticationFailure: user account has not been confirmed. Check your email for the confirmation email.
```

This will be released as 1.18.1 on PyPi today.
